CMakeLists.txt

```diff
@@ -18,6 +18,21 @@ if(NOT NO_WALL AND CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Cl
     add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
+# Set a default build type if none was specified
+set(default_build_type "Release")
+if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+    set(default_build_type "RelWithDebInfo")
```

Comment:
What's the thought behind this?

Reply:
I think the only valid choices for debug cycles are `debug` and `relwithdebinfo`. Of the two, I prefer `relwithdebinfo` as it gives a closer approximation to the actual performance during comp with release mode than `debug`. Even though `relwithdebinfo` might inline stuff, I don't think needing to drop into gdb happens _that_ often.

I took this snippet from [here](https://blog.kitware.com/cmake-and-the-default-build-type/)

Reply:
I was more talking about the `.git` check

Reply:
yeah I guess we don't really need that check